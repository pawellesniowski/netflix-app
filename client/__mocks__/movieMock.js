export default {
  "id": 269149,
  "title": "Zootopia",
  "tagline": "Welcome to the urban jungle.",
  "vote_average": 7.7,
  "vote_count": 6795,
  "release_date": "2016-02-11",
  "poster_path": "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
  "overview": "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery.",
  "budget": 150000000,
  "revenue": 1023784195,
  "genres": ["Animation", "Adventure", "Family", "Comedy"],
  "runtime": 108
}
