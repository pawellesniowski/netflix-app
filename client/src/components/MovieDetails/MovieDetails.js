import React from 'react';
import './MovieDetails.scss';

const MovieDetails = ({ selectedMovie }) => {
  const {title, release_date, poster_path, overview, runtime} = selectedMovie;
  return (
    <div className="movie-details" data-cy="movie-details">
      <div className="left-side-container">
        <img className="image" src={poster_path} alt="movie cover" data-cy="movie-details-poster"/>
      </div>
      <div className="right-side-container">
        <h2 className="title">{title}</h2>
        <h5 className="category">Oscar-winning Movies</h5>
        <span className="production-year">{release_date}</span>
        <span className="duration">{runtime ? runtime : 120} min</span>
        <p className="description">{overview}</p>
      </div>
    </div>
  )
};

export default MovieDetails;
