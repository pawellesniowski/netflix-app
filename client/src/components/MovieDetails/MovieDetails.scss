@import "../../common/styles/colors.scss";
@import "../../common/styles/mixins.scss";
@import "../../common/styles/brake-points";

.movie-details {
  @include contentContainer;
  display: flex;
  background: linear-gradient( $color-grey-black, rgba(0, 0, 0, 0.6) ), url('../../common/images/netflix-background.jpg');
  color: $color-grey-mercury;
  font-size: 3rem;
}

.left-side-container {
  padding: 2rem 2rem 2rem 0;
  .image {
    max-width: 300px;
  }
}

.right-side-container {
  padding: 4rem 2rem;
  display: block;
  min-width: 300px;
  .title {
    margin-bottom: 0.1rem;
  }
  .category {
    margin-bottom: 1rem;
  }
  .production-year {
    margin-right: 3rem;
  }
  .description {
    margin-top: 2rem;
    font-size: 1.6rem;
  }
}

@media only screen and (max-width: $mobile-resolution) {
  .movie-details {
    display: flex;
    flex-wrap: wrap;
    font-size: 2rem;
  }
  .left-side-container {
    padding: 1rem;
    .image {
      max-width: 200px;
    }
  }
  .right-side-container {
    padding: 2rem 1rem;
    display: block;
    min-width: 200px;
  }
}
