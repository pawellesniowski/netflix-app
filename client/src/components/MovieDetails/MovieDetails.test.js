import React from "react";
import {shallow} from "enzyme";
import movieMock from "../../../__mocks__/movieMock"
import MovieDetails from "./MovieDetails";


describe("MovieDetails Component", () => {
  let component;
  beforeEach(() => {
    component = shallow(<MovieDetails selectedMovie={movieMock}/>);
  });

  it("Renders the container", () => {
    expect(component.find('[data-cy="movie-details"]')).toHaveLength(1);
  });

  it("Renders all the elements", () => {
    expect(component.find('.left-side-container')).toHaveLength(1);
    expect(component.find('[data-cy="movie-details-poster"]')).toHaveLength(1);
    expect(component.find('.right-side-container')).toHaveLength(1);
    expect(component.find('.title')).toHaveLength(1);
    expect(component.find('.category')).toHaveLength(1);
    expect(component.find('.production-year')).toHaveLength(1);
    expect(component.find('.duration')).toHaveLength(1);
    expect(component.find('.description')).toHaveLength(1);
  });

  it("Matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
