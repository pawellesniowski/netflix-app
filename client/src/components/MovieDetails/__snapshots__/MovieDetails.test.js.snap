// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Component Matches the snapshot 1`] = `ShallowWrapper {}`;
